### PR TITLE
Add CI via Github actions

### PR DESCRIPTION
**Pull request by theogf**

### CODE REVIEW

**Comment by devmotion** [comment]
It would be good to update CompatHelper.yml as well such that it can trigger CI.

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -2,15 +2,26 @@ name: CompatHelper
...
+          import Pkg
+          name = "CompatHelper"
+          uuid = "aa819f21-2bde-4658-8897-bab36330d9b7"
+          version = "2"
+          Pkg.add(; name, uuid, version)
+        shell: julia --color=yes {0}
+      - name: "Run CompatHelper"
+        run: |
+          import CompatHelper
+          CompatHelper.main(; subdirs=["", "test", "test/ppl", "docs"])
```
Comment (devmotion) [inline .github/workflows/CompatHelper.yml:23]:
```suggestion
          CompatHelper.main(; subdirs=["", "test", "docs"])
```

File: .github/workflows/CompatHelper.yml [hunk 2]
```diff
@@ -2,15 +2,26 @@ name: CompatHelper
...
+        shell: julia --color=yes {0}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: julia -e 'using CompatHelper; CompatHelper.main(; subdirs = ["", "test", "docs"])'
\ No newline at end of file
+          COMPATHELPER_PRIV: ${{ secrets.DOCUMENTER_KEY }}
```
Comment (devmotion) [inline .github/workflows/CompatHelper.yml:27]: I guess the key might not exist yet?

File: .github/workflows/CompatHelper.yml (hunk 1 above)
Comment (theogf) [reply .github/workflows/CompatHelper.yml:23]: haha this what happens when you copy paste blindly, thanks for the catch

File: .github/workflows/CompatHelper.yml (hunk 2 above)
Comment (theogf) [reply .github/workflows/CompatHelper.yml:27]: I just added it

**Review by devmotion — APPROVED** [review]
Looks good! Can you also update the badge in the README?

**Pull request merged**